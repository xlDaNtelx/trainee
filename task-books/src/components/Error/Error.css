.hideError {
  display: none;
}

.showError {
  display: block; 
}
