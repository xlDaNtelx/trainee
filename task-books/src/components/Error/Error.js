import './Error.css';
import React from 'react';
import PropTypes from 'prop-types';
import { Alert } from 'reactstrap';

/**
 * Show alert if was error in get books request.
 */
class Error extends React.Component {
  render() {
    const { error } = this.props;
    return (
      <Alert color="danger" className={error ? 'showError' : 'hideError'}>
        {error}
      </Alert>
    );
  }
}

Error.propTypes = {
  /** Error that happens in request */
  error: PropTypes.string
};

Error.defaultProps = { error: '' };

export default Error;
