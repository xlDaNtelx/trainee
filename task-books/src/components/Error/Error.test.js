import React from 'react';
import renderer from 'react-test-renderer';
import Error from './index';

describe('Error', () => {
  it('renders with error in params', () => {
    const component = renderer.create(
      <Error error="Some error" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without error', () => {
    const component = renderer.create(
      <Error />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
