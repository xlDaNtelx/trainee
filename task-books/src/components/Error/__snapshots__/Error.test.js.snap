// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders with error in params 1`] = `
<div
  className="showError alert alert-danger fade"
  role="alert"
>
  Some error
</div>
`;

exports[`Error renders without error 1`] = `
<div
  className="hideError alert alert-danger fade"
  role="alert"
>
  
</div>
`;
